src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectives.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.vault.common.openam
+
+import akka.event.LoggingAdapter
+import spray.routing.Directives._
+
+object OpenAMDirectives {
+  val tokenFromCookie = cookie(OpenAMConfig.tokenCookie) map {
+    // return the content of the cookie
+    _.content
+  }
+
+  val tokenFromOptionalCookie = optionalCookie(OpenAMConfig.tokenCookie) map {
+    // when there is a cookie
+    _ map {
+      // return the content of the cookie
+      _.content
+    }
+  }
+
+  val commonNameFromCookie = tokenFromCookie map {
+    // get the common name from the token
+    case token => commonNameFromToken(token)
+  }
+
+  val commonNameFromOptionalCookie = tokenFromOptionalCookie map {
+    // when there is a cookie
+    _ map {
+      // get the common name from the token
+      case token => commonNameFromToken(token)
+    }
+  }
+
+  // Partially based off DebuggingDirectives.logRequest, sans magnets, with requestInstance instead of mapRequest
+  // TODO: Any way to make this single log parameter implicit?
+  // The magnet pattern seems to always have at least one other parameter
+  // http://spray.io/blog/2012-12-13-the-magnet-pattern/
+  def logOpenAMRequest(log: LoggingAdapter) = commonNameFromCookie flatMap { commonName =>
+    requestInstance flatMap { request =>
+      val method = request.method.name
+      val uri = request.uri
+      var message = s"$commonName $method $uri"
```

Comment:
commonName may or may not include a space, e.g. "David An" vs. "davidan". Having an unknown number of spaces makes parsing a space-delimited line difficult. Maybe make this tab-delimited? Or surround the commonName with some kind of markers?